src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeStyle/CodeStyleHelpers.cs

```diff
@@ -77,10 +77,9 @@ public static bool TryParseBoolEditorConfigCodeStyleOption(string arg, CodeStyle
             {
                 // If we have two args, then the second must be a notification option.  If 
                 // it isn't, then this isn't a valid code style option at all.
-                var secondValue = arg.Substring(firstColonIndex + 1);
-                if (TryParseNotification(secondValue, out var localNotification))
+                if (TryParseNotification(arg.AsSpan(firstColonIndex + 1), out var localNotification))
```

Comment:
no point in allocating a string when we can just use spans (which work fine with switches as well).

Reply:
Huh, didn't know you could switch on a span against strings. Neato.

Reply:
Just to make sure I understand, the compiler doesn't create the string and do the switch, but rather has the smarts to do the comparison without allocation?

Reply:
Correct.